go/src/meguca/geoip/db.go

```diff
@@ -59,3 +120,132 @@ func LookUp(ip string) (iso string) {
 
 	return
 }
+
+// check if the GeoLite2 country DB exists, and download if it doesn't
+func check() error {
+	// Get the MD5 hash from the DB and the new hash from upstream
+	oldHash, err := db.GetGeoMD5()
+
+	if err != nil {
+		return err
+	}
+
+	respMD5, err := http.Get("https://geolite.maxmind.com/download/geoip/database/GeoLite2-Country.tar.gz.md5")
+
+	if err != nil {
+		return err
+	}
+
+	defer respMD5.Body.Close()
+	md5Bytes, err := ioutil.ReadAll(respMD5.Body)
+
+	if err != nil {
+		return err
+	}
+
+	newHash := string(md5Bytes)
+
+	// Ensure the HTTP response is a MD5 hash when converted
+	if len(newHash) != 32 {
+		return errors.New("response is not an MD5 hash")
+	}
+	
+	// Load the old DB one time on server start, if applicable before checking the MD5
+	_, err = os.Stat("GeoLite2-Country.mmdb")
+	invalid := os.IsNotExist(err)
+
+	once.Do(func() {
```

Comment:
Doesn't check() only run once on server start? Do we need this?

Reply:
If you want to be able to run check() again for whatever reason, yes.
You said something about being able to do that, or something similar.

Reply:
Not sure I did. I did talk about loading the DB twice in case of have old db -> download new one.

Reply:
Either way, this ensures that if check() is ran again for some reason, we don't just load the current db each time, which would be the wrong behavior. This is the "load the current if existing and then check for new, and if new, load that one" thing.